client/src/project/new/components/Visibility.tsx

```diff
@@ -58,4 +73,209 @@ const Visibility = ({ handlers, meta, input }: VisibilityProps) => {
   );
 };
 
+interface EditVisibilityModalConfirmationProps {
+  onConfirm: (visibility: Visibilities) => void; // eslint-disable-line @typescript-eslint/ban-types
+  toggleModal: () => void; // eslint-disable-line @typescript-eslint/ban-types
+  isOpen: boolean;
+  isError: boolean;
+  isLoading: boolean;
+  isSuccess: boolean;
+  message: ReactNode;
+  visibility: Visibilities;
+}
+
+function EditVisibilityModalConfirmation({
+  isOpen,
+  toggleModal,
+  onConfirm,
+  isError,
+  isLoading,
+  isSuccess,
+  message,
+  visibility,
+}: EditVisibilityModalConfirmationProps) {
+  const buttons = !isError && !isLoading && !isSuccess && (
+    <div className="mt-2 d-flex flex-row gap-2 justify-content-end">
+      <Button
+        className="float-right mt-1 btn-outline-rk-green"
+        onClick={toggleModal}
+        data-cy="cancel-visibility-btn"
+      >
+        Cancel
+      </Button>
+      <Button
+        className="float-right mt-1 btn-rk-green"
+        onClick={() => onConfirm(visibility)}
+        data-cy="update-visibility-btn"
+      >
+        Agree
+      </Button>
+    </div>
+  );
+
+  const content =
+    isError || isSuccess ? (
+      message
+    ) : isLoading ? (
+      <LoadingLabel text="Updating visibility" />
+    ) : (
+      <>
+        Please note that users will not need your explicit permission to see
+        this project. If you need more information about visibility, please
+        check the{" "}
+        <ExternalLink
+          url={GitlabLinks.PROJECT_VISIBILITY}
+          role="text"
+          title="documentation"
+        />
+        .
+      </>
+    );
+
+  return (
+    <Modal isOpen={isOpen} toggle={() => toggleModal()} size="lg">
+      <ModalHeader toggle={() => toggleModal()}>
+        Edit visibility to{" "}
```

Comment:
```suggestion
        Change visibility to{" "}
```
Mind that this requires adjusting the e2e test -- I'll add that as a suggestion too